ui/packages/catalog-ui-search/src/main/webapp/component/search-form-list/search-form-list.view.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global require*/
+
+const CustomElements = require('../../js/CustomElements.js')
+const Marionette = require('marionette')
+const React = require('react')
+const Router = require('../router/router.js')
+const user = require('../singletons/user-instance')
+const SearchForm = require('../search-form/search-form')
+
+const NoSearchForms = () => {
+  return <div>No search forms are available</div>
+}
+
+module.exports = Marionette.ItemView.extend({
+  tagName: CustomElements.register('search-form-list'),
+  className: 'composed-menu',
+  template(props) {
+    return (
+      <React.Fragment>
+        { props.length === 0
+          ? <NoSearchForms/>
+          : null
+        }
+        {props.map(form => (
+          <div
+            key={form.id}
+            onClick={() =>
+              this.changeView(
+                new SearchForm(form),
+                this.model.get('currentQuery')
+              )
+            }
+          >
+            {form.title}
+          </div>
+        ))}
+      </React.Fragment>
+    )
+  },
+  serializeData: function() {
+    return this.model.get('searchForms').toJSON()
+  },
+  changeView: function(selectedForm, currentQuery) {
+    const sharedAttributes = selectedForm.transformToQueryStructure()
+    currentQuery.set({
+      type: 'custom',
+      ...sharedAttributes,
+    })
+    if (currentQuery.get('type') === 'custom') {
+      currentQuery.trigger('change:type')
```

Comment:
This is one of the things I was trying to prevent with this comment way back when we first introduced search forms.  https://github.com/codice/ddf/pull/3004#discussion-diff-171443373R18

Basically, we're setting ourselves up for complexity and bugs by keeping around both type and id.  We really need to address this sooner rather than later, although I fear it's probably propagated all over the place by now.  We only need id, type should be removed and the existing code depending on type refactored.

In this case, we wouldn't need to manually trigger updates.  We would instead be keying off of id, knowing that an id of 'basic' means the basic form, an id of 'advanced' means the advanced form, etc., and an id not matching those reserved keywords corresponds to the id of a custom form.  

This would also mean we wouldn't have to store two different attributes in query settings, since id could fully describe what search form we mean to show.

Reply:
I would love to clean up our use of the `type` field but I think it is out of scope for this PR